includes/Classifai/Providers/OpenAI/Embeddings.php

```diff
@@ -801,21 +809,68 @@ private function trigger_taxonomy_update( string $taxonomy = '', bool $all = fal
 			}
 		}
 
-		$args = [
+		/**
+		 * Filter the number of terms to process in a batch.
+		 *
+		 * @since 3.1.0
+		 * @hook classifai_openai_embeddings_terms_per_job
+		 *
+		 * @param {int} $number Number of terms to process per job.
+		 *
+		 * @return {int} Filtered number of terms to process per job.
+		 */
+		$number = apply_filters( 'classifai_openai_embeddings_terms_per_job', 100 );
+
+		$default_args = [
 			'taxonomy'     => $taxonomy,
 			'orderby'      => 'count',
 			'order'        => 'DESC',
 			'hide_empty'   => false,
 			'fields'       => 'ids',
 			'meta_key'     => 'classifai_openai_embeddings', // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
 			'meta_compare' => 'NOT EXISTS',
-			'number'       => $this->get_max_terms(),
+			'number'       => $number,
+			'offset'       => 0,
 			'exclude'      => $exclude, // phpcs:ignore WordPressVIPMinimum.Performance.WPQueryParams.PostNotIn_exclude
 		];
 
+		$default_args = array_merge( $default_args, $args );
+
 		// If we want all terms, remove our meta query.
 		if ( $all ) {
-			unset( $args['meta_key'], $args['meta_compare'] );
+			unset( $default_args['meta_key'], $default_args['meta_compare'] );
+		} else {
+			unset( $default_args['offset'] );
+		}
+
+		if ( 0 === $user_id ) {
+			$user_id = get_current_user_id();
+		}
+
+		$job_args = [
+			'taxonomy' => $taxonomy,
+			'all'      => $all,
+			'args'     => $default_args,
+			'user_id'  => $user_id,
+		];
+
+		\as_enqueue_async_action( 'classifai_schedule_generate_embedding_job', $job_args );
```

Comment:
One thing to consider. If we have no terms to generate embeddings for, we still schedule this action and show a message to the user that generation is in progress. This means anytime the settings are saved, that message shows. Wondering if we want to run the actual term query here and then only proceed if we have results?